### PR TITLE
Add tectonic default ranks

### LINKED ISSUE #5294: Add default tectonic ranks
There is no special logic or requirements for this tree, and the default should simply be a root node and 5 simple ranks:

Superstructure
Tectonic domain
Tectonic subdomain
Tectonic unit
Tectonic subunit

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #5294

### Checklist

- [ ] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [ ] Add relevant issue to release milestone

### Testing instructions

- Open Tectonic Tree
- Verify the ranks below are present
- - [ ] Superstructure
- - [ ] Tectonic domain
- - [ ] Tectonic subdomain
- - [ ] Tectonic unit
- - [ ] Tectonic subunit
- [ ] Verify there is a root node

### CODE REVIEW

File: specifyweb/specify/models_by_table_id.py [hunk 1]
```diff
@@ -208,7 +208,7 @@
     1017: 'Collectionobjectgroupjoin',
     1018: 'Collectionobjectgrouptype',
 }
-
+#TODO: do we need to add tectnocUnit to this file?
```
Comment (CarolineDenis) [inline specifyweb/specify/models_by_table_id.py:211]: add age tables and tectonic tables. 
Order has an importance?

File: specifyweb/specify/tree_ranks.py [hunk 2]
```diff
@@ -77,13 +77,26 @@
     'member': 400,
     'bed': 500
 }
+TECTONIC_UNIT_RANKS = {
+    'root': 0,
+    'superstructure': 10,
+    'tectonic_domain': 20,
+    'tectonic_subdomain': 30,
+    'tectonic_unit': 40,
```
Comment (CarolineDenis) [inline specifyweb/specify/tree_ranks.py:85]: which of the two format do we need?

File: specifyweb/stored_queries/execution.py [hunk 3]
```diff
@@ -85,6 +85,14 @@ def filter_by_collection(model, query, collection):
         logger.info("filtering geologic time period rank to discipline: %s", collection.discipline.name)
         return query.filter(model.GeologicTimePeriodTreeDefID == collection.discipline.geologictimeperiodtreedef_id)
 
+    if model is models.TectonicUnit:
+        logger.info("filtering tectonic unit to discipline: %s", collection.discipline.name)
+        return query.filter(model.TectonicUnitTreeDefID == collection.discipline.tectonicunittreedef_id)
+
+    if model is models.TectonicUnitTreeDefItem:
```
Comment (CarolineDenis) [inline specifyweb/stored_queries/execution.py:92]: this line prevents from querying for tectonic unit item on form when adding the def item in the relationship

**Review by sharadsw — APPROVED** [review]

**Review by combs-a — APPROVED** [review]
### Testing instructions
- Open Tectonic Tree
- Verify the ranks below are present
- - [X] Superstructure
- - [X] Tectonic domain
- - [X] Tectonic subdomain
- - [X] Tectonic unit
- - [X] Tectonic subunit
- [X] Verify there is a root node 

Tested on ChadronTest. All the ranks and the root node was there, looks good.

image

**Review by grantfitzsimmons — CHANGES_REQUESTED** [review]
Looks good in the app, but please add the migration explanation and other metadata to the [**Migrations Description**](https://docs.google.com/document/d/1a74C7NnAXtkynn0KAHVd6g7lXXj_n-GJ1LS68IYqjAs/edit?tab=t.0#heading=h.bhqvg1cfzp5r)

We should try to document these as we go along! Probably best to even link to the PR.

Note that this migration will need to be recreated when new disciplines are added in the future.

The only issue I encountered was reverting this migration. See below:
```sh
specify@260f0e593b10:/opt/specify7$ ve/bin/python manage.py showmigrations specify

specify
 [X] 0001_initial
 [X] 0002_geo
 [X] 0003_cotype_picklist
 [X] 0004_stratigraphy_age
 [X] 0005_collectionobjectgroup_parentcojo
 [X] 0006_fix_tectonic_tree_fields
 [X] 0007_tectonic_ranks
specify@260f0e593b10:/opt/specify7$ 
specify@260f0e593b10:/opt/specify7$ ve/bin/python manage.py migrate specify 0006_fix_tectonic_tree_fields

Operations to perform:
  Target specific migration: 0006_fix_tectonic_tree_fields, from specify
Running migrations:
  Rendering model states... DONE
  Unapplying specify.0007_tectonic_ranks...Traceback (most recent call last):
  File "manage.py", line 25, in <module>
    execute_from_command_line(sys.argv)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/core/management/commands/migrate.py", line 244, in handle
    post_migrate_state = executor.migrate(
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/db/migrations/executor.py", line 121, in migrate
    state = self._migrate_all_backwards(plan, full_plan, fake=fake)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/db/migrations/executor.py", line 196, in _migrate_all_backwards
    self.unapply_migration(states[migration], migration, fake=fake)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/db/migrations/executor.py", line 252, in unapply_migration
    state = migration.unapply(state, schema_editor)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/db/migrations/migration.py", line 174, in unapply
    operation.database_backwards(self.app_label, schema_editor, from_state, to_state)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/db/migrations/operations/special.py", line 196, in database_backwards
    self.reverse_code(from_state.apps, schema_editor)
  File "/opt/specify7/specifyweb/specify/migrations/0007_tectonic_ranks.py", line 118, in revert_cosolidated_python_django_migration_operations
    revert_default_tectonic_ranks(apps, schema_editor)
  File "/opt/specify7/specifyweb/specify/migrations/0007_tectonic_ranks.py", line 64, in revert_default_tectonic_ranks
    tectonic_tree_def.delete()
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/db/models/query.py", line 745, in delete
    collector.collect(del_query)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/db/models/deletion.py", line 302, in collect
    raise ProtectedError(
django.db.models.deletion.ProtectedError: ("Cannot delete some instances of model 'TectonicUnitTreeDef' because they are referenced through protected foreign keys: 'TectonicUnitTreeDefItem.treedef', 'TectonicUnit.definition'.", {<TectonicUnitTreeDefItem: TectonicUnitTreeDefItem object (1)>, <TectonicUnitTreeDefItem: TectonicUnitTreeDefItem object (2)>, <TectonicUnitTreeDefItem: TectonicUnitTreeDefItem object (3)>, <TectonicUnitTreeDefItem: TectonicUnitTreeDefItem object (4)>, <TectonicUnitTreeDefItem: TectonicUnitTreeDefItem object (5)>, <TectonicUnitTreeDefItem: TectonicUnitTreeDefItem object (6)>, <TectonicUnit: TectonicUnit object (1)>})
specify@260f0e593b10:/opt/specify7$ 
```

**Review by grantfitzsimmons — CHANGES_REQUESTED** [review]
```
specify@f494bdad9830:/opt/specify7$ ve/bin/python manage.py migrate specify 0006_fix_tectonic_tree_fields
Operations to perform:
  Target specific migration: 0006_fix_tectonic_tree_fields, from specify
Running migrations:
  Rendering model states... DONE
  Unapplying specify.0007_tectonic_ranks...Traceback (most recent call last):
  File "manage.py", line 25, in <module>
    execute_from_command_line(sys.argv)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/core/management/commands/migrate.py", line 244, in handle
    post_migrate_state = executor.migrate(
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/db/migrations/executor.py", line 121, in migrate
    state = self._migrate_all_backwards(plan, full_plan, fake=fake)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/db/migrations/executor.py", line 196, in _migrate_all_backwards
    self.unapply_migration(states[migration], migration, fake=fake)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/db/migrations/executor.py", line 252, in unapply_migration
    state = migration.unapply(state, schema_editor)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/db/migrations/migration.py", line 174, in unapply
    operation.database_backwards(self.app_label, schema_editor, from_state, to_state)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/db/migrations/operations/special.py", line 196, in database_backwards
    self.reverse_code(from_state.apps, schema_editor)
  File "/opt/specify7/specifyweb/specify/migrations/0007_tectonic_ranks.py", line 118, in revert_cosolidated_python_django_migration_operations
    revert_default_tectonic_ranks(apps, schema_editor)
  File "/opt/specify7/specifyweb/specify/migrations/0007_tectonic_ranks.py", line 64, in revert_default_tectonic_ranks
    tectonic_tree_def.delete()
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/db/models/query.py", line 745, in delete
    collector.collect(del_query)
  File "/opt/specify7/ve/lib/python3.8/site-packages/django/db/models/deletion.py", line 302, in collect
    raise ProtectedError(
django.db.models.deletion.ProtectedError: ("Cannot delete some instances of model 'TectonicUnitTreeDef' because they are referenced through protected foreign keys: 'TectonicUnitTreeDefItem.treedef', 'TectonicUnit.definition'.", {<TectonicUnitTreeDefItem: TectonicUnitTreeDefItem object (1)>, <TectonicUnitTreeDefItem: TectonicUnitTreeDefItem object (2)>, <TectonicUnitTreeDefItem: TectonicUnitTreeDefItem object (3)>, <TectonicUnitTreeDefItem: TectonicUnitTreeDefItem object (4)>, <TectonicUnitTreeDefItem: TectonicUnitTreeDefItem object (5)>, <TectonicUnitTreeDefItem: TectonicUnitTreeDefItem object (6)>, <TectonicUnit: TectonicUnit object (1)>})
specify@f494bdad9830:/opt/specify7$ 
What's next:
    Try Docker Debug for seamless, persistent debugging tools in any container or image → docker debug specify7-specify7-1
    Learn more at https://docs.docker.com/go/debug-cli/
```

@CarolineDenis Could you revert successfully?

**Review by grantfitzsimmons — APPROVED** [review]
```
~/GitHub/specify7 on issue-5316 *1 !2 ❯ docker exec -it specify7-specify7-1 /bin/bash                                                                          Py specify7 at 04:01:11 PM
specify@82862d65616e:/opt/specify7$ ve/bin/python manage.py migrate specify 0006_fix_tectonic_tree_fields
Operations to perform:
  Target specific migration: 0006_fix_tectonic_tree_fields, from specify
Running migrations:
  Rendering model states... DONE
  Unapplying specify.0007_tectonic_ranks... OK
specify@82862d65616e:/opt/specify7$ 
```

**Review by acwhite211 — APPROVED** [review]

**Pull request merged**